urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/httpd/CustomUtil.java

```diff
@@ -0,0 +1,536 @@
+package ikrs.httpd;
+
+
+import java.io.File;
+import java.io.InputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import ikrs.typesystem.*;
+import ikrs.util.CustomLogger;
+import ikrs.util.CaseInsensitiveComparator;
+
+/**
+ * This class holds some helper methods.
+ *
+ * @author Ikaros Kappler
+ * @date 2012-09-11
+ * @version 1.0.0
+ *
+ **/
+
+
+public class CustomUtil {
+
+
+
+    public static boolean getBoolean( CustomLogger logger,
+				      BasicType wrapper,
+				      boolean nullSubstitute ) {
+	
+	if( wrapper == null )
+	    return nullSubstitute;
+	else
+	    return wrapper.getBoolean();
+    }
+
+
+    public static String getFileExtension( File file ) 
+	throws NullPointerException {
+	
+	int index = file.getName().lastIndexOf(".");
+	if( index == -1 || index == 0 )
+	    return null;
+	else if( index == file.getName().length()-1 )
+	    return "";
+	else
+	    return file.getName().substring( index+1, file.getName().length() );
+	
+    }
+
+    public static String processCustomizedFilePath( String filePath ) {
+	if( filePath == null )
+	    return null;
+
+	filePath = filePath.replaceAll( "\\{USER_HOME\\}", System.getProperty("user.home") );
+	// ...
+	
+	return filePath;
+    }
+
+    public static boolean equalFileExtensions( String extensionA,
+					       String extensionB,
+					       boolean caseSensitive,
+					       boolean dotSensitive ) {
+
+	if( extensionA == extensionB )
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.